api_app/services/aad_authentication.py

```diff
@@ -282,27 +282,42 @@ def _get_app_auth_info(self, client_id: str) -> dict:
 
         return authInfo
 
-    def _get_role_assignment_graph_data_for_user(self, user_id: str) -> dict:
+    def _ms_graph_query(self, url: str, http_method: str, json=None) -> dict:
         msgraph_token = self._get_msgraph_token()
+        auth_headers = self._get_auth_header(msgraph_token)
+        graph_data = {}
+        while True:
+            if not url:
+                break
+            logging.debug(f"Making request to: {url}")
+            if json:
+                response = requests.request(method=http_method, url=url, json=json, headers=auth_headers)
+            else:
+                response = requests.request(method=http_method, url=url, headers=auth_headers)
+            url = ""
+            if response.status_code == 200:
```

Comment:
What happens in the error case here?
Prior to this change, I think the error response was returned to the caller

Reply:
only the payload was returned and the callers are looking for the "value" property which will be missing anyway.
Should I just throw an exception here?

Reply:
In the non-success case previously, I think the payload contained the error details (and callers logged the payload if `value` wasn't found).
With the changes, it looks like the non-success case now returns an empty payload?

Reply:
I added a log. The caller is throwing the exception.